### PR TITLE
feat: update icon based on state

### LINKED ISSUE #56: Dynamically update extension icon
The extension icon should show states like disabled, shown or hidden.

### PR DESCRIPTION
**Pull request by rofe**
Fix #56

### CODE REVIEW

File: src/extension/tab.js [hunk 1]
```diff
@@ -62,6 +61,9 @@ async function getProxyUrl({ id, url: tabUrl }) {
  * @param {Object[]} matches The config matches
  */
 async function injectContentScript(tabId, matches) {
+  if (!await chrome.tabs.get(tabId)) {
+    return;
+  }
```
Comment (rofe) [inline src/extension/tab.js:66]: This should hopefully eliminate the frequent
```
Unchecked runtime.lastError: No tab with id: <id>.
```
errors in the service worker.

File: src/extension/manifest.json [hunk 2]
```diff
@@ -63,11 +63,11 @@
     "action": {
       "default_title": "__MSG_title__",
       "default_icon": {
-        "16": "icons/extension-icon-16x16.png",
-        "32": "icons/extension-icon-32x32.png",
-        "48": "icons/extension-icon-48x48.png",
-        "128": "icons/extension-icon-128x128.png",
-        "512": "icons/extension-icon-512x512.png"
+        "16": "icons/disabled/icon-16x16.png",
```
Comment (rofe) [inline src/extension/manifest.json:66]: The default icon state is `disabled` and will be changed based on URL/state.

File: src/extension/ui.js [hunk 3]
```diff
@@ -22,12 +24,51 @@ let updateInProgress = false;
 
 /**
  * @typedef {Object} Context
- * @prop {number} id The tab ID
- * @prop {string} url The URL
+ * @prop {string} [url] The URL
  * @prop {Object} [config] The project config
+ * @prop {Object[]} [matches] The config matches
```
Comment (dylandepass) [inline src/extension/ui.js:29]: Should this be [OptionsConfig](https://github.com/adobe/aem-sidekick/blob/main/src/extension/types/typedefs.js#L35) ?

File: src/extension/ui.js [hunk 4]
```diff
@@ -10,8 +10,10 @@
  * governing permissions and limitations under the License.
  */
 
+// import log from './log.js';
```
Comment (dylandepass) [inline src/extension/ui.js:13]: remove?

File: src/extension/ui.js [hunk 5]
```diff
@@ -22,12 +24,51 @@ let updateInProgress = false;
...
+export async function updateIcon({ matches = [] }) {
+  let iconType = 'disabled';
+  if (matches.length > 0 && isValidProject(matches[0])) {
+    if (await getDisplay()) {
+      iconType = 'default';
+    } else {
+      iconType = 'hidden';
+    }
+  }
+  // log.debug(`updateIcon: using icon type ${iconType}`);
```
Comment (dylandepass) [inline src/extension/ui.js:65]: remove?

**Review by dylandepass — APPROVED** [review]

File: src/extension/ui.js (hunk 3 above)
Comment (rofe) [reply src/extension/ui.js:29]: Possibly 😉

File: src/extension/ui.js (hunk 4 above)
Comment (rofe) [reply src/extension/ui.js:13]: Forgot to readd 😉

File: src/extension/ui.js (hunk 5 above)
Comment (rofe) [reply src/extension/ui.js:65]: Forgot to readd 😉

**Pull request merged**